Integrations: Add `mssql` integration (#2795)
grafana/agent: docs/sources/configuration/integrations/mssql-config.md

---
title: mssql_config
---

# mssql_config

> The `mssql_configs` block configures the `mssql` integration,
> which is an embedded version of [`sql_exporter`](https://github.com/burningalchemist/sql_exporter).
> This allows the collection of [Microsoft SQL Server](https://www.microsoft.com/en-us/sql-server) metrics.

Reviewer: ```suggestion
The `mssql_configs` block configures the `mssql` integration, an embedded version of [`sql_exporter`](https://github.com/burningalchemist/sql_exporter) that lets you collect [Microsoft SQL Server](https://www.microsoft.com/en-us/sql-server) metrics.
```

> It is recommended to have a dedicated user set up for the purpose of monitoring an mssql instance.
> The user for monitoring must have the following grants in order to populate the metrics:

Reviewer: ```suggestion
It is recommended that you have a dedicated user set up for monitoring an mssql instance.
The user for monitoring must have the following grants in order to populate the metrics:
```

```
GRANT VIEW ANY DEFINITION TO <MONITOR_USER>
GRANT VIEW SERVER STATE TO <MONITOR_USER>
```

Full reference of options:

```yaml
  # Enables the mssql integration, allowing the Agent to automatically
  # collect metrics for the specified mssql instance.
  [enabled: <boolean> | default = false]

  # Sets an explicit value for the instance label when the integration is
  # self-scraped. Overrides inferred values.
  #
  # The default value for this integration is the host:port of the provided connection_string.
  [instance: <string>]

  # Automatically collect metrics from this integration. If disabled,

>   # the snowflake integration is run but not scraped and thus not

Reviewer: Should this be snowflake?

Author: Nope, good catch!
